core/src/Kokkos_Macros.hpp

```diff
@@ -172,7 +172,10 @@
 
 #define KOKKOS_IMPL_HIP_CLANG_WORKAROUND
 
+#define HIP_ENABLE_PRINTF
+#define HCC_ENABLE_PRINTF
 #include <hip/hip_runtime.h>
+#include <hc_printf.hpp>
```

Comment:
I don't think this is still necessary with rocm 3.1, is it?

Reply:
Yes, but you need it for `rocm-3.0`. I am fine with removing it.

Reply:
I think we should only support the latest version of ROCM given the state of the HIP compiler...